### PR TITLE
Check that reads don't exceed the maximal length

### PR DESCRIPTION
**Pull request by csoneson**
Also allow reads up to 65534 nt

### CODE REVIEW

**Comment by mbstadler** [comment]
I think the check for having read a read that is too long is great!

Maybe we have to reduce the default read-length, though, or indeed expose it as a parameters, to avoid memory issues.
For example, the re-building of the vignette block starting at L174 fails under Linux on GHA with `std::bad_alloc`, indicating that it could not allocate requested memory. My guess is that this happened here:
https://github.com/fmicompbio/mutscan/blob/130c85296bfda8be69737991ec5cadf72ff8f05b/src/digestFastqs.cpp#L1149

which will try to allocate a `FastqBuffer` instance with `BUFFER_SIZE * chunksize * x` bytes (x is 2 for single-red and 4 for paired-end data), which will here be around 64kb * 1e5 * 4 =~ 26GB (probably not available on the GitHub action runner). We could reduce sequence length, chunksize, or both. What about `BUFFER_SIZE = 4096` and keep the current chunksize = 100000? That would allocate about 1.6GB for a paired end experiment.

**Comment by csoneson** [comment]
Yeah, seems I went a bit over the top (worked fine locally 🙂). A read length of 4096 will not be enough for the ONT data, so maybe we indeed do need to expose it as a parameter.

**Comment by mbstadler** [comment]
Right, in that case an exposed parameter and a lower default value would probably be best.

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# mutscan 0.2.36
+
+* Check that reads don't exceed the maximal allowed length
+* Allow read lengths up to 65534 nt
```
Comment (mbstadler) [inline NEWS.md:4]: I guess here we could say `* Add parameter to specify maximal read length` instead of the second item

**Review by mbstadler — APPROVED** [review]
I think this is a neat solution that leaves open all options, and the defaults are likely to cover the most common cases without creating a too large memory footprint. Apart from the comment on the `NEWS.md` file, I have no suggestions. Thank you!

**Comment by mbstadler** [comment]
The current failing checks on windows might be related to https://github.com/r-lib/actions/issues/713 and should resolve itself soon

File: NEWS.md (hunk 1 above)
Comment (csoneson) [reply NEWS.md:4]: Oops, yes, definitely - forgot to update this.

**Pull request merged**